src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py

```diff
@@ -845,6 +846,46 @@ def test_download_run(self, pf: PFClient, created_batch_run_without_llm: Run):
             for file in expected_files:
                 assert Path(tmp_dir, created_batch_run_without_llm.name, file).exists()
 
+    @pytest.mark.usefixtures("mock_isinstance_for_mock_datastore")
+    def test_upload_run(
+        self,
+        pf: PFClient,
+        randstr: Callable[[str], str],
+        subscription_id: str,
+        resource_group_name: str,
+        workspace_name: str,
+    ):
+        from promptflow._sdk._pf_client import PFClient as LocalPFClient
+
+        trace_provider = (
+            f"azureml://subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/"
+            f"providers/Microsoft.MachineLearningServices/workspaces/{workspace_name}"
+        )
+        with patch.object(Configuration, "get_trace_provider", return_value=trace_provider):
+            name = randstr("batch_run_name")
```

Comment:
according to current replay mechanism, name will always be `batch_run_name` in replay mode, so when run this test more than once, this test will fail as this run is already there; not sure if this can be refined from test itself, or need some changes from replay

Reply:
maybe add a pre-step to delete this run first?

Reply:
as offline synced, i will help add a try delete